### PR TITLE
Feature/136 setup token authentication

### PR DESCRIPTION
**Pull request by duhizjame**

### CODE REVIEW

File: server/app/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,2 +1,6 @@
 io.delta.sharing.api.server.defaultMaxResults=10
-quarkus.banner.path=banner.txt
\ No newline at end of file
+quarkus.banner.path=banner.txt
+
+whitefox.server.authentication.enabled=false
+whitefox.server.authentication.bearerToken=myToken
```
Comment (tmnd1991) [inline server/app/src/main/resources/application.properties:5]: I think these should stay in the test application properties, not the the main one, am I right?

Comment (duhizjame) [reply server/app/src/main/resources/application.properties]: I would keep the whitefox.server.authentication.enabled=false as default and remove the token one

File: server/app/src/main/java/io/whitefox/api/server/auth/SimpleTokenAuthenticationMechanism.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Override
+  public Uni<ChallengeData> getChallenge(RoutingContext context) {
+    return null;
+  }
+
+  @Override
+  public Set<Class<? extends AuthenticationRequest>> getCredentialTypes() {
+    return null;
+  }
```
Comment (tmnd1991) [inline server/app/src/main/java/io/whitefox/api/server/auth/SimpleTokenAuthenticationMechanism.java:58]: are those fine to be null?

File: server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * <ul>
+ *   <li>Prevents the Quarkus OIDC authentication mechanism from attempting authentication when it
+ *       is not configured. Note that attempts to use the OIDC authentication mechanism when the
+ *       authentication server is not properly configured will result in 500 errors as opposed to
+ *       401 (not authorized).
+ *   <li>Completely disallows unauthenticated requests when authentication is enabled.
+ * </ul>
+ */
+@Alternative // @Alternative + @Priority ensure the original HttpAuthenticator bean is not used
```
Comment (tmnd1991) [inline server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java:30]: <3

File: server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+@Priority(1)
+@Singleton
+public class WhitefoxHttpAuthenticator extends HttpAuthenticator {
+
+  @Inject
+  IdentityProviderManager identityProvider;
```
Comment (tmnd1991) [inline server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java:36]: can't this be private final as the others?

File: server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  @Override
+  public Uni<SecurityIdentity> attemptAuthentication(RoutingContext context) {
+    if (!authEnabled) {
+      return anonymous();
+    }
+    // quarkus dev paths
+    else if (context.normalizedPath().startsWith("/q/")) {
+      return anonymous();
+    } else
+      return selectAuthenticationMechanism(config, context).authenticate(context, identityProvider);
```
Comment (tmnd1991) [inline server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticator.java:72]: nit: let's add braces around this too

File: server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxAuthenticationConfig.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  @WithName("enabled")
+  @WithDefault("false")
+  boolean enabled();
+
+  @WithName("bearerToken")
+  Optional<String> bearerToken();
+
+  /** Returns the set of HTTP URL paths that are permitted to be serviced without authentication. */
+  @WithName("anonymous-paths")
+  Set<String> anonymousPaths();
```
Comment (tmnd1991) [inline server/app/src/main/java/io/whitefox/api/server/auth/WhitefoxAuthenticationConfig.java:23]: Are these really used?

File: server/app/src/test/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticatorTest.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  public static class NoAuthenticationProfile implements QuarkusTestProfile {
+    @Override
+    public Map<String, String> getConfigOverrides() {
+      return Map.of("whitefox.server.authentication.enabled", "false");
+    }
+  }
+
+  // Nesting two test-profiles leads to an OMM: https://github.com/quarkusio/quarkus/issues/12498
```
Comment (tmnd1991) [inline server/app/src/test/java/io/whitefox/api/server/auth/WhitefoxHttpAuthenticatorTest.java:49]: nit: OOM
```suggestion
  // Nesting two test-profiles leads to an OOM: https://github.com/quarkusio/quarkus/issues/12498
```

**Review by tmnd1991 — CHANGES_REQUESTED** [review]

File: server/app/src/main/resources/application.properties [hunk 8]
```diff
@@ -1,2 +1,5 @@
 io.delta.sharing.api.server.defaultMaxResults=10
-quarkus.banner.path=banner.txt
\ No newline at end of file
+quarkus.banner.path=banner.txt
+
+whitefox.server.authentication.enabled=${WHITEFOX_SERVER_AUTHENTICATION_ENABLED:-false}
```
Comment (tmnd1991) [inline server/app/src/main/resources/application.properties:4]: you don’t need to set the property to the environment variable, quarkus does that automatically :) so you can remove the next line and simplify this one

File: docsite/docs/architecture/authentication.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# Whitefox authentication
+
+By default, whitefox runs as a insecure server.
+
+## Token authentication
+
+We provide a simple way to secure your application using a token provided as an environment variable.
+
+Two properties from application.properties are used to control the authentication methods of your instance:
```
Comment (tmnd1991) [inline docsite/docs/architecture/authentication.md:9]: i would create a table with the property names and expected values and point to quakus doc about property resolution :)

**Pull request merged**